feel/feeldiscr/operatorinterpolation.hpp

```diff
@@ -1074,30 +1258,26 @@ OperatorInterpolation<DomainSpaceType, ImageSpaceType,IteratorRange,InterpType>:
 
                                 for ( uint16_type jloc = 0; jloc < domain_basis_type::nLocalDof; ++jloc )
                                 {
-                                    uint16_type compDomain = (domain_basis_type::is_product)? comp : 0;
+                                    uint16_type nCompDomain = (domain_basis_type::is_product)?domain_basis_type::nComponents:1;
+                                    for(int cdomain = 0;cdomain < nCompDomain;++cdomain )
+                                    {
```

Comment:
the row of the matrix represent the interpolant at a components.
you add in the row all components contribution of the domain function.
It's still block diagonal but the coupled block are non null.
```
v_x(p) = u^1_x*id(u^1(p))_x + u^2_x*id(u^2(p))_x + ... + u^1_y*id(u^1(p))_y + u^2_y*id(u^2(p))_y
with u^i_x the value of domain function at component x at the i-ême local dof point
and id(u^i)_x the value of i eme test function at point p and for the component x
```
I need to look more. I am not sure that all combinaison are done in testsuite.